src/createServer.js

```diff
@@ -1,3 +1,42 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+'use strict';
+
+const http = require('http');
+const { convertToCase } = require('./convertToCase');
+const { parseUrl } = require('./parseUrl.js');
+const { validateUrl } = require('./validateUrl');
+
+function createServer() {
+  const server = http.createServer((req, res) => {
+    res.setHeader('Content-Type', 'application/json');
+
+    const [textToConvert, toCase] = parseUrl(req);
+    const errors = validateUrl(textToConvert, toCase);
+
+    if (errors.length > 0) {
+      res.statusCode = 400;
+      res.statusMessage = 'Bad request';
+      res.end(JSON.stringify({ errors }));
+    } else {
+      res.statusCode = 200;
+      res.statusMessage = 'OK';
+
+      const { originalCase, convertedText }
+      = convertToCase(textToConvert, toCase);
+
+      const jsonResponse = {
+        originalCase,
+        targetCase: toCase,
+        originalText: textToConvert,
+        convertedText,
+      };
+
+      res.end(JSON.stringify(jsonResponse));
+    }
```

Comment:
Is it necessary to wrap this code in `else` construction?
You can use it as a default case.

```suggestion
    } 
    res.statusCode = 200;
    res.statusMessage = 'OK';

    const { originalCase, convertedText } = convertToCase(textToConvert, toCase);

    const jsonResponse = {
       originalCase,
       targetCase: toCase,
       originalText: textToConvert,
       convertedText,
    };

    res.end(JSON.stringify(jsonResponse));
```